opentech/apply/determinations/templatetags/determination_tags.py

```diff
@@ -0,0 +1,60 @@
+from django import template
+from django.core.exceptions import ObjectDoesNotExist
+
+from opentech.apply.determinations.models import UNDETERMINED
+from opentech.apply.funds.workflow import DETERMINATION_PHASES
+
+register = template.Library()
+
+
+@register.filter
+def is_lead_or_admin(user, submission):
+    return submission.user_lead_or_admin(user)
+
+
+@register.filter
+def can_add_determination(user, submission):
```

Comment:
I think there is too much going on here. These functions are taking on multiple responsibilities and  duplicating checks that are being done elsewhere.

e.g. `has_determination_draft` does a permission check and a draft check

We share a lot of this functionality and the views check similar things. we are going to get quite confused if functionality changes and the view checks one thing and the template tags check another.